### ISSUE TITLE
Operate can't complete backup snapshot

### ISSUE THREAD

**Issue by ralfpuchert** [camunda/camunda#18049]
### Describe the bug

When Operate takes a backup it schedules several snapshots (currently 6) for it. Then it checks for every snapshot the state for progress. In case that the snapshot state doesn't change for any reason (e.g. it's running exceptionally long; bug in Elasticsearch), then the check is running forever without any further information for the user.

### To Reproduce

* We cannot fully reproduce this scenario at the moment
* See [Support 21530](https://jira.camunda.com/browse/SUPPORT-21530)

### Current behavior

* After the create snapshot request initially times out, we check the snapshot status periodically via ES API and enter into a loop until the snapshot status `"IN PROGRESS"` resolves (see https://github.com/camunda/camunda/blob/operate-8.5.6/operate/webapp/src/main/java/io/camunda/operate/webapp/elasticsearch/backup/ElasticsearchBackupRepository.java#L405)
* More details in [Support 21530](https://jira.camunda.com/browse/SUPPORT-21530)

### Expected behavior

* We make transparent to the user what is happening: add logs to periodically report that the snapshot is still in progress (TODO: decide on log level and frequency); this should include the snapshot ID so that the user can double-check against the Elasticsearch API
* We introduce a configurable timeout so that
  * the snapshot status polling ends once exceeded
  * the snapshot request fails, so that the user can clean up and decide if they want to try it again
  * The default value for that timeout is 0 (unbounded; users need to opt in) to avoid that we introduce regressions for users where the snapshot procedure takes long but is generally successful

### Environment
Seen in versions 8.3.2 and above

- OS: [e.g. MacOS]
- Browser: [e.g. chrome, safari]
- Operate Version: [e.g. 8.1]

### Additional context

---------------------------------------------------------------------------------------------

### Definition of Ready - Checklist

- [ ] The bug has been reproduced by the assignee in the environment compatible with the provided one; otherwise, the issue is closed with a comment
- [ ] The issue has a meaningful title, description, and testable acceptance criteria
- [ ] The issue has been labeled with an appropriate `Bug-area` label
- [ ] Necessary screenshots, screen recordings, or files are attached to the bug report

For UI changes required to solve the bug:

- [ ] Design input has been collected by the assignee

### :point_right: Handover Dev to QA

- Changed components:
- Side effects on other components:
- Handy resources:
  BPMN/DMN models, plugins, scripts, REST API endpoints + example payload, etc :
  
- Example projects:

- Commands/Steps needed to test; Versions to validate:

- Docker file / HELM chart : in case that it needed to be tested via docker share the version contain the fixed along with version of other services .

- Release version ( in which version this fixed/feature will be released):

**Comment by alxgrk** [comment]
> The check for snapshot status = "IN PROGRESS" for "part 5" can run forever.

For the sake of completeness, I observed the same issue on version 8.2.26.

**Comment by alxgrk** [comment]
@ralfpuchert tried a few more times and still not getting past the 5th part. Any idea what could be a workaround to proceed with creating a backup for now?

**Comment by alxgrk** [comment]
I found a workaround: for me, setting the `max_snapshot_bytes_per_sec` on the Elasticsearch S3 repository to "100mb" (default is "40mb") was at least enough to have the 5th part running slightly faster and finish within the default Elasticsearch socket timeout of 30 seconds. See also https://www.elastic.co/guide/en/elasticsearch/reference/current/repository-s3.html#repository-s3-repository & https://www.elastic.co/guide/en/elasticsearch/client/java-rest/6.8/_timeouts.html.

I also tried to dig a little deeper into the code that is responsible for handling failures when creating a snapshot & it looks like the reason why it is failing for me on 8.2.26 is that the retry logic for `SocketTimeoutException`s was only introduced on 2023/08/28 with https://github.com/camunda/camunda/commit/63c987003ca9ea20d007aeb2893032129a764c7a, which should be version 8.5.0.

**Comment by ThorbenLindhauer** [comment]
Hi @alxgrk,

the handling of timeout exceptions was first released with 8.3.0 and backported and released in patches 8.1.17 and 8.2.16. As ther Operate code and issues for those versions are closed source I cannot share the evidence for that. Note that on the PR that you found, 2023/08/28 is before the 8.3.0 release date of 2023/10/08. The reason why this version is not showing on the commit is that these Operate versions live in a different repository.

Accordingly, I think your explanation is not the rootcause of your issue. If you have a way to reproduce the problem and can share the exact steps with us, then that would really help us.

Cheers,
Thorben

**Comment by ThorbenLindhauer** [comment]
I have updated the ticket description to describe what exactly we will change with implementing it.

**Comment by alxgrk** [comment]
Thanks @ThorbenLindhauer for the explanation. I also saw your comment on the PR, answering here. I'm not sure what version of the code I viewed, but since my assumption was that the problem came from the missing retry mechanism, I tried to increase the timeout in that case. Of course, this is not a proper solution, but it was my approach to providing a workaround for older versions and decreasing the probability of running into this issue.
As I said, slightly speeding things up by increasing the bytes/sec limit helped me to make the 5th part finish in less than 30 secs.

Reproducing is probably hard, since all I can provide is what I already wrote (the version, the Elasticsearch repository configured in S3 and that I can see from the logs that the 5th part of the snapshot times out after 30 seconds).
